descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,308 @@
+
+        return validation_data
+
+    def run_on_single_catalog(self, galaxy_catalog, catalog_name, base_output_dir):
+        
+        #get catalog data
+        mag_field = galaxy_catalog.first_available(*self.possible_mag_fields)
+        if not mag_field:
+            return TestResult(skipped=True, summary='Missing requested quantities')
+        catalog_data = self.get_catalog_data(galaxy_catalog, [self.z, mag_field], filters=self.filters)
+        #filtername = mag_field.partition(self.band + '_')[-1].upper() doesn't work for g band!
+        filtername = re.split(self.band + '_',mag_field)[-1].upper()
+        filelabel = '_'.join([filtername,self.band])
+
+        fig, ax  =  plt.subplots(self.nrows, self.ncolumns, figsize=(figx_p,figy_p), sharex='col')
+        self.yaxis = 'P(z|m)' if self.normed else 'N(z|m)'
+
+        catalog_color = next(self.colors)
+        
+        #loop over magnitude cuts and make plots
+        results = {}
+        for n, (ax_this, summary_ax_this, cut_lo, cut_hi, z0, z0err) in enumerate(zip_longest(
+                ax.flat,
+                self.summary_ax.flat,
+                self.mag_lo,
+                self.mag_hi,
+                self.validation_data.get('z0values', []),
+                self.validation_data.get('z0errors', []),
+        )):
+            if cut_lo is not None:
+                mask = (catalog_data[mag_field] < cut_lo)
+                cutlabel = '{} $< {}$'.format(self.band, cut_lo)
+                if cut_hi:
+                    mask &= (catalog_data[mag_field] >= cut_hi)
+                    cutlabel = '${} <=$ '.format(cut_hi) + cutlabel #also appears in txt file so don't use \leq
+
+                if z0 is None and 'z0const' in self.validation_data.keys():
+                    z0 = self.validation_data['z0const'] + self.validation_data['z0linear'] * cut_lo
+            
+                z_this = catalog_data[self.z][mask]
+                del mask
+                total = '(# of galaxies = {})'.format(len(z_this))
+
+                #bin catalog_data
+                N,binEdges = np.histogram(z_this, bins=self.zbins)
+                sumz,binEdges = np.histogram(z_this, bins=self.zbins,weights=z_this)
+                meanz = sumz/N
+
+                #make subplot
+                ncol = int(n%self.ncolumns)
+                nrow = n//self.ncolumns
+                catalog_label = ' '.join([catalog_name, re.sub(self.band,filtername+' '+self.band,cutlabel)])
+                validation_label = ' '.join([self.validation_data.get('label', ''), cutlabel])
+                reskey = re.sub('\$','',cutlabel)
+                results[reskey] = self.make_subplot(meanz, z_this, n, ax_this, z0, z0err, catalog_color, validation_color, catalog_label, validation_label)
+                results[reskey]['total'] = total
+
+                #add curve for this catalog to summary plot
+                if self.first_pass:  #add validation data if evaluating first catalog
```

Comment:
Use `None` to be more explicit.  Remove `summary = ` since it is not used.
```python
if self.first_pass:  #add validation data if evaluating first catalog
    self.make_subplot(meanz, z_this, n, summary_ax_this, z0, z0err, catalog_color, validation_color, catalog_label, validation_label)
else:
    self.make_subplot(meanz, z_this, n, summary_ax_this, None, None, catalog_color, validation_color, catalog_label, validation_label)
```

Reply:
Done